### PR TITLE
docgen: update pulpMd (+ transitively, markdownfmt) to fix markdown bugs

### PR DESCRIPTION
**Pull request by karel-3d**
I have put pulpMd and markdownfmt to better shape (and added some tests there); now let's use the new versions.

Also fixed invisible space in annex template

Also fixed master test failure, related to recent docgen changes

### CODE REVIEW

**Comment by karel-3d** [comment]
For some reason, I cannot add reviewers here, so /cc @favadi

File: docgen/templates/annex.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 {{range $m := .Messages}}
 
-## {{$m.Name}}
+## {{$m.Name}}
```
Comment (karel-3d) [inline docgen/templates/annex.md]: these are tiny changes, that should be a separate commits; but I didn't want to make 2 commits, because changing the test scripts is so painful every time :/

File: testdata/scripts/extract.txt [hunk 2]
```diff
@@ -552,15 +552,17 @@ msgstr ""
...
 msgstr ""
 -- all.md.golden --
-# Accounts API v1.0.0
+Accounts API v1.0.0
+===================
 
 Provides CRUD operations on the accounts resource.
 
-* Host `https://openbank.com`
+*   Host `https://openbank.com`
```
Comment (karel-3d) [inline testdata/scripts/extract.txt]: markdownfmt now uses more spaces, because that is sometimes necessary by blackfriday (a kind-of buggy markdown parser, that we use in both markdownfmt and hugo)

File: testdata/scripts/extract.txt [hunk 3]
```diff
@@ -632,12 +635,12 @@ curl -X POST \
...
-| Name       | Type   | Description                                                                |
-|------------|--------|----------------------------------------------------------------------------|
-| account_id | string | AccountID is the unique identifier of the newly created account.           |
-| number     | string | CreditTransactionNumber is the transaction number of the credited account. |
-| number     | string | DebitTransactionNumber is the transaction number of the debited account.   |
-| minor      | string | TODO: add comment.                                                         |
+| Name                      | Type   | Description                                                                |
+|---------------------------|--------|----------------------------------------------------------------------------|
+| account_id                | string | AccountID is the unique identifier of the newly created account.           |
+| credit_transaction_number | string | CreditTransactionNumber is the transaction number of the credited account. |
```
Comment (karel-3d) [inline testdata/scripts/extract.txt]: this was an actual bug in markdownfmt

File: testdata/scripts/extract.txt (hunk 2 above)
Comment (favadi) [reply testdata/scripts/extract.txt]: that looks weird, do you know why blackfriday requires double spaces here?

File: testdata/scripts/extract.txt [hunk 4]
```diff
@@ -690,7 +696,8 @@ curl -X DELETE \
 | 403    | Returned when the user does not have permission to access the resource.                |
 | 500    | Returned when an unexpected error occured on the server side.                          |
 
-## Retrieve an account
+Retrieve an account
+-------------------
```
Comment (favadi) [inline testdata/scripts/extract.txt]: why it changes from ## to ______?

File: testdata/scripts/extract.txt (hunk 2 above)
Comment (karel-3d) [reply testdata/scripts/extract.txt]: Blackfriday is not commonmark compliant in weird ways. One of the weirdest issues is the spacing of lists

https://github.com/russross/blackfriday/issues/329 (3 years old issue now)

This does not parse correctly
```
* foo
  * bar
    * baz
```

this does
```
*   foo
    *   bar
        *   baz
```

this does not parse correctly
```
1. A paragraph

   item continues
```

this does (1 extra space)

```
1.  A paragraph

    item continues
```

Comment (karel-3d) [reply testdata/scripts/extract.txt]: Hugo switched to GoldMark some time ago, but not the version we are using.

File: testdata/scripts/extract.txt (hunk 4 above)
Comment (karel-3d) [reply testdata/scripts/extract.txt]: Because I thought it's more "common" way to write headers in markdown, so markdownfmt converts `## foo` to 

```
foo
---
```

(original markdownfmt did that too, just our fork didn't)

**Comment by karel-3d** [comment]
can you please merge when you think it's OK? I don't have write access here :(

**Comment by favadi** [comment]
I would like to have @Kunde21 take a look at this before merging.

**Comment by karel-3d** [comment]
btw I just fixed markdownfmt+pulpMd to use goldmark instead of blackfriday... our (old) hugo version still uses blackfriday though, so I still format lists with this wider spacing

File: assets/generated/annex.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 {{range $m := .Messages}}
 
-## {{$m.Name}}
+## {{$m.Name}}
```
[the remark this answers is not in the document]
Comment (karel-3d) [reply assets/generated/annex.md]: The original file had an invisible space (`&nbsp;`), probably as a mistake, This is a change to a normal space.

[the remark this answers is not in the document]
Comment (mvdan) [reply assets/generated/annex.md]: SGTM. The commit message should probably clarify this.

[the remark this answers is not in the document]
Comment (karel-3d) [reply assets/generated/annex.md]: I *can* break it into more smaller commits if needed, I was just too lazy to

[the remark this answers is not in the document]
Comment (mvdan) [reply assets/generated/annex.md]: No need, I think. It's just that, seeing the diff in plaintext, this change is confusing.

**Comment by karel-3d** [comment]
Updated to new pulpMd version

Also update commit message to clarify the comments here (the markdown changes + the invisible space)

File: testdata/scripts/extract.txt (hunk 2 above)
Comment (karel-3d) [reply testdata/scripts/extract.txt]: I now switched our Hugo to the version with Goldmark, so I guess I can change it back so it uses normal two spaces :)

Comment (karel-3d) [reply testdata/scripts/extract.txt]: I switched to Hugo everywhere in internal docs, so this is done now

**Comment by karel-3d** [comment]
finally!

@favadi maybe this messes up your PRs elsewhere, sorry for that, but finally merging
